python_scripts/trees_classification.py

```diff
@@ -92,13 +99,28 @@
 print(f"Accuracy of the LogisticRegression: {test_score:.2f}")
 
 # %% [markdown]
-# Unlike linear models, decision trees are non-parametric models: they are not
-# controlled by a mathematical decision function and do not have weights or
+# Unlike linear models, the decision rules for the decision trees are not
+# controlled by a mathematical decision function and do not have weights or an
 # intercept to be optimized.
```

Comment:
This sentence is still not correct: a decision tree prediction function can still be written an a mathematical expression (e.g. a sum of products with boolean indicator terms for the split nodes and weights for the leaf nodes). However the number of terms and trainable "weights" is not fixed ahead of time as for linear models.

The following is more correct:


```suggestion
# Unlike linear models, the decision rule for the decision tree is not
# controlled by a simple linear combination of weights and feature values.
```